docs/aws/audit/opensearchmonitoring/rules/opensearch_in_vpc_only_remediation.mdx

```diff
@@ -1,6 +1,256 @@
+
+2. In the navigation pane, choose "Domains". This will display a list of all your OpenSearch domains.
+
+3. Click on the name of the domain you want to check. This will open the domain dashboard where you can see all the configurations for that domain.
+
+4. In the domain dashboard, look for the "VPC" section. If the domain is inside a VPC, you will see the VPC ID, Subnet ID, and Security Group ID. If these fields are empty or not present, then the OpenSearch domain is not in a VPC.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by downloading the appropriate installer from the AWS website. Once installed, you can configure it by running `aws configure` and providing your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all OpenSearch domains: Use the `aws es list-domain-names` command to list all the OpenSearch (formerly known as Elasticsearch) domains. This command will return a list of all the domain names.
+
+   Command: 
+   ```
+   aws es list-domain-names
+   ```
+   This will return a list of domain names in the following format:
+   ```
+   {
+       "DomainNames": [
+           {
+               "DomainName": "example-domain"
+           },
+           ...
+       ]
+   }
+   ```
+
+3. Describe the OpenSearch domain: For each domain, use the `aws es describe-elasticsearch-domain --domain-name <DomainName>` command to get detailed information about the domain.
+
+   Command: 
+   ```
+   aws es describe-elasticsearch-domain --domain-name example-domain
+   ```
+   This will return a JSON object with detailed information about the domain.
+
+4. Check if the domain is in a VPC: In the output of the previous command, look for the `VPCOptions` field. If this field is present and the `VPCId` field is not null, then the OpenSearch domain is in a VPC. If the `VPCOptions` field is not present or the `VPCId` field is null, then the OpenSearch domain is not in a VPC.
+
+   Example of a domain in a VPC:
+   ```
+   "VPCOptions": {
+       "VPCId": "vpc-1a2b3c4d",
+       ...
+   }
+   ```
+   Example of a domain not in a VPC:
+   ```
+   "VPCOptions": {
+       "VPCId": null,
+       ...
+   }
+   ```
+   or
+   ```
+   "VPCOptions": {}
+   ```
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. This can be done using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Establish a session: You need to establish a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'region_name' with your AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>